### DISCUSSION TITLE
Fixing griefing for good: boredem, accountability, mitigation.

### DISCUSSION THREAD

**Discussion by Cogs16** [FakeFishGames/Barotrauma#5742] [Ideas and suggestions]
**TL:DR** Griefers grief because of boredem and a lack of accountability. Add cosmetics, activities, chains of command, and a global commendation(a la overwatch) system to stop this.

Currently Barotrauma casual mode suffers from a serious griefing problem. I think there are three main causes of griefing, boredom, an inability in game to mitigate griefing, and a lack of accountability. If these three issues were to be fixed, I think there could be a massive reduction griefing.

**Boredom:**
Right now people grief because it is fun, and because often times in submarines there is nothing to do. Sure when monsters attack, theres something to do thats fun. Unfortunately, on lower difficulties this only happens once in a while. Unless the crew is doing some good roleplay, or coordinating, there is often nothing to do between attacks. I think if there were more roleplay opportunities, people would be less bored and less inclined to grief. In order to create these opportunities, I think all that has to be done is the following:
	**-More cosmetics**
		-Add factional uniforms, hats, rich people clothing, poor people clothing, chefs uniforms (even if there are no chefs) let people be what they want to be. Want to roleplay as a hobo? Take the ragged uniform. Want to roleplay the rich CEO ordering the crew around? Grab the suit.
	**-Board games**
		Some board games such as chess or tic tac toe would be fun to play during off times, additionally it would also add hilarity as the submarine sinks.
	**-Possibly a more indepth repair system**
		Right now repairing things means walking around and clicking on them. At the moment, it is not very rewarding. If repairs say involved minigames and components it could make repairing items much more rewarding. Additionally, it could also replace the skill level system with genuine knowledge, which adds depth to the game. (I.e I am faster because I know how to repair pumps, not because of an arbitrary number)

**Inability to mitigate griefing:**
At the moment, griefers cannot often be stopped in game by other players and the damage they do is often irreversible. Security cannot stun, or easily move people about the ship to detain them without killing them, and often killing innocents in the process. Additionally, determining who a griefer is without seeing them is nigh impossible unless you have a console log. Lastly, if a griefer unwires the submarine, there is no way to determine where the wires need to go. Solutions:
	**-Add auto-generated circuit diagrams**
		These would be pamphlets that contain the default wiring information for the ship, every engineer would have them
	**-Bot auto rewiring commands**
		Give bots an innate knowledge of the ships wiring so that if asked they can return the wiring, to its default.
	**-Security DNA scanner**
		A DNA scanner that tells security who touched what and when would enable security to find griefers, and also make the game more interesting for them.
	**-Tandem carrying**
		Suggested earlier in another post, make it possible for two people to carry a ragdoll faster, so that griefers can be dragged to the brig where they will be secure.
	

	**Accountability:** Lastly, griefing occurs because there is no accountability, both for the people trying to stop them and the griefer themselves. If there were more long-term consequences for griefers, or a way to ensure a functional chain of command to mitigate them, then the griefing could see another dramatic reduction. A few possible ideas:
	**-Add a global commendation system to barotrauma.** 
		Everyone in the game would be able to give one upvote to another player to improve their reputation. If another player does things well, you can commend them. Server owners would then be able to set commendation limits, so as to keep people with low to no commendation out. While there would be a lot of details with this system to work out, the basic idea of a consistent reputation could be used to combat griefing. It could also lead to medals which player characters could display (i.e good service medals)
	**-Add chains of accountability or improve the control the captain has over security**
		-Right now no professional force can be organized in casual games consistently because security can be anyone regardless of if they follow orders. A possible solution to this is to add the option for a vote, wherein the captain of the ship is chosen by the crew, and then the captain can appoint a set number of security officers. This would enable the captain to maintain a proper chain of command, as they could choose secs who would listen to them. However, the captain also has to act like a captain, or else get voted off by the crew.
	**-Overhaul the security bots**
		-Security bots should be able to enforce specific orders and be asked to detain specific people. That way the captain could maintain control of the ship.

These are just some ideas, but I think they would have massive impacts on gameplay for the better

**Comment by thegrb93** [comment]
Griefing is part of the fun. Use the karma system if you want a super serious grief free server.

**Reply by Cogs16** [reply]
Karma has alot of room for improvement, also griefing IS part of the fun. But policing should also be part of the fun, and right now sec doesn't really have the tools to do it, and again too much of anything is bad for gameplay.

**Reply by PrincessMuu-Muu** [reply]
I personally hate when people play security on my server thinking its a moderator/admin role. Even more so when they start dictating who does what and then I have to teach them that I'm the owner the hard way. 

The idea of giving security mock logs would only truly benefit the game during traitor mode. Even then, I still feel the methods for the traitors to complete their objectives is already extremely limited. Traitor mode itself needs a revamp before things like this are added. 

I agree with @thegrb93 just because a lot of the things mentioned are already possible on a serious/roleplay server with the right settings and admins in play. 

(I also believe the devs want the individual classes to be more valuable and required to play, so I don't see them adding minigame repairs. I've never supported that motion from the devs, as many people do not have 3-5+ friends to play with.)

**Reply by Cogs16** [reply]
1) I see things differently. I think security does exist to police things internally, they exist to stop mutiny, they exist to protect the ship from both internal threats and external ones. I think their duties do overlap somewhat with moderators(however are also differing), as both exist to make the game "playable". In my view, both have the power to moderate the game, except one exists within the game lore wise, and one exists outside of it. I guess my question to you is if they don't do internal policing that then what do they do exactly, also does that mean no griefing at atall?

2) I think a security and traitor revamp are not mutually exclusive and need to go hand in hand. Both need serious reworks, and I think both can happen and even need to happen at the same time since they play off each other. 

3) Karma is inorganic, and not nearly customizable enough. Karma also eliminates securities role, as they no longer need and also CANNOT (since they get space herpes) enforce internal order. They are rendered redundant as often times a ship has enough crewmembers of the other classes to man the guns, or pic up weapons to fight monsters. Additionally, it does not help with the accountability problem, as there is no way for the captain to "make" people do their jobs in casual games. (I know baro is supposed to be cooperative and assumes cooperation, but there still needs to be a system of accountability at some level, even if that meant adding more loosing conditions) If karma could be made more flexible, so that say it affects different classes differently, or even integrated with the security system (I.e instead of herpes, people are marked as traitors for guards to then deal with) that would be more organic. But right now its inflexible and destroys immersion, when there are other more organic solutions that breed more opportunities for interesting gameplay.

4) I don't really understand this point, as people play with bots without friends, and also play on smaller subs instead of larger ones.

3)

**Reply by MystLeissa** [reply]
> I personally hate when people play security on my server thinking its a moderator/admin role. Even more so when they start dictating who does what and then I have to teach them that I'm the owner the hard way.
> 
> The idea of giving security mock logs would only truly benefit the game during traitor mode. Even then, I still feel the methods for the traitors to complete their objectives is already extremely limited. Traitor mode itself needs a revamp before things like this are added.
> 
> I agree with @thegrb93 just because a lot of the things mentioned are already possible on a serious/roleplay server with the right settings and admins in play.
> 
> (I also believe the devs want the individual classes to be more valuable and required to play, so I don't see them adding minigame repairs. I've never supported that motion from the devs, as many people do not have 3-5+ friends to play with.)

Personally? You'd have to be pretty inexperienced to think Security was any kind of admin role, anyone thinking that is probably just griefing you to begin with. Sad part is it's entirely possible for miscommunication to happen, but ultimately it's up to the individual server what they do about griefing. I know I won't play on Public Servers because of this and other communications issues. 

The Real Job of Security, is to defend the sub, and it's crew. Traitors are (relatively) rare, so way I see it everyone has a few mistakes to make. Ordering people around iis not exactly a good idea unless it's in your "lane". Of course, Job Roles gets into a whole different can of worms. I just say scrap this suggestion (OP) as a whole, and take a more isolated approach. Security is for manning the guns, and keeping husks off my ship, anything else a medical officer can do, and even those tasks a MD can do relatively well. :P

**Comment by fracter2** [comment]
I agree with many of the ideas here, but especially with the idea of having preparing stuff being a more skill-based minigame rather than clicking and waiting for a timer to end. that goes beyond the scope of just griefing.

**Comment by ViktorShield** [comment]
Chess in Barotrauma?
Shut up and take my money!

What about wiring, I feel like adding a small magazine attached to the right part of the wiring interface would be a good improvement. Clicking on it would open it, so you see how were wires attached at the start of the round. In traitors mode. It might also make traitor and engineer jobs more interesting.

**Comment by Ek-Videogames** [comment]
Obligatory: *'If you institute global banlists allow us to ban admins for being shit'*

**Comment by xSideburns** [comment]
Generally find the stun baton doesn't stun for long enough to get the handcuffs on. Increasing the stun baton effect time might also assist security dealing with issues. Or perhaps giving security the ability to restrain a person when grabbing them effectively making them limp. But like the stun baton with a chance of failing perhaps based on a comparison of each individuals security / shooting / combat skills?

Then again a lot of griefers show up as security : 3 Perhaps an option that means the captain or host of a server has to approve people wanting a security role? Maybe everyone starts as an assistant and has to work their skills into the role they want, once more promotion from assistant into specific role is approved by captain / host?

**Comment by Cogs16** [comment]
Just putting this here since the accountability part got formatted weirdly.

**Accountability:** Lastly, griefing occurs because there is no accountability, both for the people trying to stop them and the griefer themselves. If there were more long-term consequences for griefers, or a way to ensure a functional chain of command to mitigate them, then the griefing could see another dramatic reduction. A few possible ideas:
**-Add a global commendation system to barotrauma.** 
	Everyone in the game would be able to give one upvote to another player to improve their reputation. If another player does things well, you can commend them. Server owners would then be able to set commendation limits, so as to keep people with low to no commendation out. While there would be a lot of details with this system to work out, the basic idea of a consistent reputation could be used to combat griefing. It could also lead to medals which player characters could display (i.e good service medals)
**-Add chains of accountability or improve the control the captain has over security**
	-Right now no professional force can be organized in casual games consistently because security can be anyone regardless of if they follow orders. A possible solution to this is to add the option for a vote, wherein the captain of the ship is chosen by the crew, and then the captain can appoint a set number of security officers. This would enable the captain to maintain a proper chain of command, as they could choose secs who would listen to them. However, the captain also has to act like a captain, or else get voted off by the crew.
**-Overhaul the security bots**
	-Security bots should be able to enforce specific orders and be asked to detain specific people. That way the captain could maintain control of the ship.

**Comment by ubertpendragon** [comment]
Adding virtual tic tac toe won't do anything against griefers, who are first and foremost joining to fuck your sub up. I've seen countless griefers simply throwing away items from the sub, replacing fuel tanks with oxygen, or randomly revoming wires. Is that any less boring that actively participating in the missions?

Griefers are NOT a part of the game, they're not fun to deal with, they can singlehandedly ruin a campaign. 

You do not "deal" with griefers using game mechanics, you do it with meta actions: reading logs, banning, and cheating
I have to constantly play with cheats on to undo whatever plague they unleash on my sub. 

Hosting a public server means having to constantly scan logs, waste time and effort combating griefers, there is no such thing as "fun" in griefer hunting and It's not possible for "security" effectively prevent griefers from deleting all of your morphine, rewiring important parts of the sub or randomly stabbing people. Especially when most griefers play security themselves for the free SMG

If you want any anti-griefing system to be efficient, it has to be:

1. Preventive, immediately take action before the griefer is allowed to do anything on the server
2. Global, so the game itself has a database for known griefers

Start counting the bans across servers, and gate the griefers from participating, either through a hard lock (preventing griefers from joining) or a soft lock (preventing griefers from acting in-game)

You don't reform griefers by giving them toys, you reform them by punishing. Some will actually try to play the game. Others will quit playing Barotrauma if they can't grief, and that's a net positive
